### PR TITLE
feat(conditionBuilder): accessibility and generic issue fixes

### PR DESCRIPTION
**Pull request by amal-k-joy**
Contributes #4916 , #5640 

- Implementation to show content as '**incomplete**' when user close the popover without selecting any values
- Edge case issue when user delete all plain condition keeping only subgroups in a group
- Mange tab index and focus
- Accessibility fixes

#### How did you test and verify your work?
Local storybook

### CODE REVIEW

File: packages/ibm-products/src/components/ConditionBuilder/ConditionGroupBuilder/ConditionGroupBuilder.js [hunk 1]
```diff
@@ -45,12 +46,46 @@ const ConditionGroupBuilder = ({
...
-        ),
-      });
+
+      if (group.conditions[1].conditions) {
+        //when we remove every plain conditions of a group without deleting  the subgroup, we need to restructure the group
+        //the inner group become outer group and same level subgroups become plain conditions
+
+        let newGroup = [],
+          allConditions = [];
+        group.conditions.map((eachItem) => {
```
Comment (davidmenendez) [inline packages/ibm-products/src/components/ConditionBuilder/ConditionGroupBuilder/ConditionGroupBuilder.js:56]: instead of using `map` you can use `reduce` similarly to the way you do below to simplify this
```
const allConditions = group.conditions.reduce((acc, item) => {
  if (item.conditions) {
    return acc.concat(item.conditions);
  }
  return acc;
}, []);
```

File: packages/ibm-products/src/components/ConditionBuilder/ConditionGroupBuilder/ConditionGroupBuilder.js [hunk 2]
```diff
@@ -45,12 +46,46 @@ const ConditionGroupBuilder = ({
...
+          if (eachItem.conditions) {
+            allConditions = [...allConditions, ...eachItem.conditions];
+          } //spreading out the condition inside the subgroup
+        });
+        //we always have conditions first and then subgroups, so ordering accordingly
+        const groupedItems = allConditions.reduce((acc, item) => {
```
Comment (davidmenendez) [inline packages/ibm-products/src/components/ConditionBuilder/ConditionGroupBuilder/ConditionGroupBuilder.js:62]: we can eliminate some redundant code by modifying the checks
```
const groupedItems = allConditions.reduce((acc, item) => {
  if (item.conditions) {
    if (!acc.groups) {
      acc.groups = [];
    }
    acc.groups.push(item);
  } else {
    if (!acc.conditions) {
      acc.conditions = [];
    }
    acc.conditions.push(item);
  }
  return acc;
}, {});
```

File: packages/ibm-products/src/components/ConditionBuilder/ConditionGroupBuilder/ConditionGroupBuilder.js [hunk 3]
```diff
@@ -45,12 +46,46 @@ const ConditionGroupBuilder = ({
...
-        conditions: group.conditions.filter(
-          (condition) => conditionId !== condition.id
-        ),
-      });
+
+      if (group.conditions[1].conditions) {
+        //when we remove every plain conditions of a group without deleting  the subgroup, we need to restructure the group
+        //the inner group become outer group and same level subgroups become plain conditions
+
+        let newGroup = [],
```
Comment (davidmenendez) [inline packages/ibm-products/src/components/ConditionBuilder/ConditionGroupBuilder/ConditionGroupBuilder.js:54]: since `newGroup` isn't actually used till line 73 i would just move it's instantiation to there. `const newGroup = [ ... ]`

File: packages/ibm-products/src/components/ConditionBuilder/ConditionGroupBuilder/ConditionGroupBuilder.js [hunk 4]
```diff
@@ -45,12 +46,46 @@ const ConditionGroupBuilder = ({
...
+            acc['groups'] = acc['groups'] || [];
+            acc['groups'].push(item);
+          } else {
+            acc['conditions'] = acc['conditions'] || [];
+            acc['conditions'].push(item);
+          }
+          return acc;
+        }, {});
+
+        newGroup.conditions = [
```
Comment (davidmenendez) [inline packages/ibm-products/src/components/ConditionBuilder/ConditionGroupBuilder/ConditionGroupBuilder.js:73]: if i'm reading this correctly it looks like this is just `allConditions`. if that's the case then `groupedItems` really inst necessary at all since you can just do `newGroup.conditions = [].concat(allConditions)`. although i would like to point out that you're using property assignment on an array here, which should most likely be done on an object instead.

**Review by davidmenendez — CHANGES_REQUESTED** [review]
few comments for ya

File: packages/ibm-products/src/components/ConditionBuilder/ConditionBuilderItem/ConditionBuilderItem.js [hunk 5]
```diff
@@ -22,30 +24,30 @@ export const ConditionBuilderItem = ({
...
+  const { conditionBuilderRef } = useContext(ConditionBuilderContext);
   const getPropertyDetails = () => {
-    if (label === 'INVALID') {
+    const { property, operator } = condition || {};
+    if (
+      label === 'INVALID' ||
+      (rest['data-name'] === 'propertyField' && property === 'INVALID') ||
+      (rest['data-name'] === 'operatorField' && operator === 'INVALID')
+    ) {
+      label = 'INVALID';
```
[the remark this answers is not in the document]
Comment (amal-k-joy) [reply packages/ibm-products/src/components/ConditionBuilder/ConditionBuilderItem/ConditionBuilderItem.js]: thanks to notify this. Resolved.

File: packages/ibm-products/src/components/ConditionBuilder/ConditionGroupBuilder/ConditionGroupBuilder.js (hunk 1 above)
Comment (amal-k-joy) [reply packages/ibm-products/src/components/ConditionBuilder/ConditionGroupBuilder/ConditionGroupBuilder.js]: modified as advised

File: packages/ibm-products/src/components/ConditionBuilder/ConditionGroupBuilder/ConditionGroupBuilder.js (hunk 2 above)
Comment (amal-k-joy) [reply packages/ibm-products/src/components/ConditionBuilder/ConditionGroupBuilder/ConditionGroupBuilder.js]: Did some overall refactor in the logic.

File: packages/ibm-products/src/components/ConditionBuilder/ConditionGroupBuilder/ConditionGroupBuilder.js (hunk 4 above)
Comment (amal-k-joy) [reply packages/ibm-products/src/components/ConditionBuilder/ConditionGroupBuilder/ConditionGroupBuilder.js]: Actually its just `allConditions` . But allConditions will contain normal conditions as well as any subgroups. So we need rearrange `allConditions` in such a way that conditions first and then any subgroups. This is an edge scenario when user delete all conditions in a group. 

https://github.com/carbon-design-system/ibm-products/assets/153802538/4e1e3043-e79d-499e-8c70-512ffd151f92

File: packages/ibm-products/src/components/ConditionBuilder/ConditionGroupBuilder/ConditionGroupBuilder.js (hunk 3 above)
Comment (amal-k-joy) [reply packages/ibm-products/src/components/ConditionBuilder/ConditionGroupBuilder/ConditionGroupBuilder.js]: resolved

**Review by davidmenendez — APPROVED** [review]
awesome!

**Review by makafsal — APPROVED** [review]
